scripts/protoc

```diff
@@ -5,12 +5,34 @@ _dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 _protoc_version="4.0.0-rc2"
 _protoc_version_filename="4.0.0-rc-2"
 _protoc_path="/tmp/pomerium-protoc/protoc-$_protoc_version"
+_protoc_3pp_path="/tmp/pomerium-protoc-3pp"
 _os="linux"
 if [ "$(uname -s)" == "Darwin" ]; then
   _os="osx"
 fi
 
-mkdir -p "$_protoc_path"
+
+# fetch_zip dir name url
+download() {
+  local dst=$1/$2
+  if [ -d $dst ]; then
+    return
+  fi
+
+  echo downloading $2
+  local archive=`mktemp`
+  curl -L -o $archive $3
+  mkdir -p $dst
+  tar xzf $archive -C $dst --strip-components=1
+  rm ${archive}
+}
+
+download $_protoc_3pp_path protoc-gen-validate https://github.com/envoyproxy/protoc-gen-validate/tarball/v0.4.1
```

Comment:
We should probably have and use a variables for these (e.g. like `_protoc_version` and so on)

Reply:
or migrate to Bazel that has a proper release pinning with hash validation like 

```bazel
http_archive(
    name = "io_bazel_rules_go",
    sha256 = "7904dbecbaffd068651916dce77ff3437679f9d20e1a7956bff43826e7645fcc",
    urls = [
        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.25.1/rules_go-v0.25.1.tar.gz",
        "https://github.com/bazelbuild/rules_go/releases/download/v0.25.1/rules_go-v0.25.1.tar.gz",
    ],
)
```